timeseries/src/autogluon/timeseries/models/gluonts/torch/models.py

```diff
@@ -0,0 +1,181 @@
+
+    def _get_hpo_backend(self):
+        return CUSTOM_BACKEND
+
+    def _get_estimator_init_args(self) -> Dict[str, Any]:
+        """Get GluonTS specific constructor arguments for estimator objects, an alias to
+        `self._get_model_params` for better readability."""
+        init_kwargs = self._get_model_params()
+
+        # GluonTS does not handle context_length=1 well, and sets the context to only prediction_length
+        # we set it to a minimum of 10 here.
+        init_kwargs["context_length"] = max(10, init_kwargs.get("context_length", self.prediction_length))
+
+        return init_kwargs
+
+    def _get_estimator(self) -> GluonTSPyTorchLightningEstimator:
+        """Return the GluonTS Estimator object for the model"""
+
+        # As GluonTSPyTorchLightningEstimator objects do not implement `from_hyperparameters` convenience
+        # constructors, we re-implement the logic here.
+        # we translate the "epochs" parameter to "max_epochs" for consistency in the AbstractGluonTSModel
+        # interface
+
+        init_args = self._get_estimator_init_args()
+
+        trainer_kwargs = {}
+        epochs = init_args.get("max_epochs", init_args.get("epochs"))
+        callbacks = init_args.get("callbacks", [])
+
+        if epochs is not None:
+            trainer_kwargs.update({"max_epochs": epochs})
+        trainer_kwargs.update({"callbacks": callbacks, "enable_progress_bar": False})
+
+        return from_hyperparameters(
+            self.gluonts_estimator_class,
+            trainer_kwargs=trainer_kwargs,
+            **init_args,
+        )
+
+    def _fit(
+        self,
+        train_data: TimeSeriesDataFrame,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        time_limit: int = None,
+        **kwargs,
+    ) -> None:
+        verbosity = kwargs.get("verbosity", 2)
+        set_logger_verbosity(verbosity, logger=logger)
+        gts_logger.setLevel(logging.ERROR if verbosity <= 3 else logging.INFO)
+        for pl_logger in pl_loggers:
+            pl_logger.setLevel(logging.ERROR if verbosity <= 3 else logging.INFO)
+
+        if verbosity > 3:
+            logger.warning(
+                "GluonTS logging is turned on during training. Note that losses reported by GluonTS "
+                "may not correspond to those specified via `eval_metric`."
+            )
+
+        self._check_fit_params()
+        # TODO: reintroduce early stopping callbacks
+
+        # update auxiliary parameters
+        callbacks = [Timer(timedelta(seconds=time_limit))] if time_limit is not None else []
+        self._deferred_init_params_aux(dataset=train_data, callbacks=callbacks, **kwargs)
+
+        estimator = self._get_estimator()
+        with warning_filter(), disable_root_logger():
+            self.gts_predictor = estimator.train(
+                self._to_gluonts_dataset(train_data),
+                validation_data=self._to_gluonts_dataset(val_data),
+            )
```

Comment:
It seems that the difference between `AbstractGluonTSPyTorchModel._fit` and `AbstractGluonTSMXNetModel._fit` is only related to how we set up the callbacks. Maybe we can move the callback setup to a separate method and have a single `_fit` method in `AbstractGluonTSModel`?